tests/unit/Administration/Login/Config/LoginConfigServiceTest.php

```diff
@@ -0,0 +1,313 @@
+            ],
+
+            'redirect_uri is no a url' => [
+                'config' => [
+                    'use_default' => false,
+                    'client_id' => 'clientId',
+                    'client_secret' => 'clientSecret',
+                    'redirect_uri' => 'redirectUri',
+                    'base_url' => 'http://base.url',
+                ],
+                'exceptionMessage' => 'Login config is incomplete or misconfigured. Field errors: [redirect_uri] is invalid URL',
+            ],
+
+            'base_url is not set' => [
+                'config' => [
+                    'use_default' => false,
+                    'client_id' => 'clientId',
+                    'client_secret' => 'clientSecret',
+                    'redirect_uri' => 'http://redirect.url',
+                ],
+                'exceptionMessage' => 'Login config is incomplete or misconfigured. Field errors: [base_url] is missing',
+            ],
+
+            'base_url is null' => [
+                'config' => [
+                    'use_default' => false,
+                    'client_id' => 'clientId',
+                    'client_secret' => 'clientSecret',
+                    'redirect_uri' => 'http://redirect.url',
+                    'base_url' => null,
+                ],
+                'exceptionMessage' => 'Login config is incomplete or misconfigured. Field errors: [base_url] is null, [base_url] is blank',
+            ],
+
+            'base_url is blank' => [
+                'config' => [
+                    'use_default' => false,
+                    'client_id' => 'clientId',
+                    'client_secret' => 'clientSecret',
+                    'redirect_uri' => 'http://redirect.url',
+                    'base_url' => '',
+                ],
+                'exceptionMessage' => 'Login config is incomplete or misconfigured. Field errors: [base_url] is blank',
+            ],
+
+            'base_url is no a string' => [
+                'config' => [
+                    'use_default' => false,
+                    'client_id' => 'clientId',
+                    'client_secret' => 'clientSecret',
+                    'redirect_uri' => 'http://redirect.url',
+                    'base_url' => 12,
+                ],
+                'exceptionMessage' => 'Login config is incomplete or misconfigured. Field errors: [base_url] is invalid string, [base_url] is invalid URL',
+            ],
+
+            'base_url is no a url' => [
+                'config' => [
+                    'use_default' => false,
+                    'client_id' => 'clientId',
+                    'client_secret' => 'clientSecret',
+                    'redirect_uri' => 'http://redirect.url',
+                    'base_url' => 'baseUrl',
+                ],
+                'exceptionMessage' => 'Login config is incomplete or misconfigured. Field errors: [base_url] is invalid URL',
+            ],
+        ];
+    }
+}
```

Comment:
What about testing the other two public methods of `LoginConfigService`:
- createTemplateData
- createRedirectUrl